### DISCUSSION TITLE
json object from const char*

### DISCUSSION THREAD

**Discussion by Kolhun** [nlohmann/json#4130] [Q&A] [answered]
There is a section of code that cannot be viewed through printf, but it is possible through std::cout, how can this be done?

![image](https://github.com/nlohmann/json/assets/36815965/6004217f-96f9-4d5b-97bc-bcd48c9598f9)
![image](https://github.com/nlohmann/json/assets/36815965/cbfde15c-04c2-42d1-b997-8e3e4837c072)

**Comment by nlohmann** [comment, marked as answer]
`operator[]` returns a reference to `basic_json` which is nothing `print` can print. Try `array["name"].get_ref<std::string&>.c_str()`.
